src/scf/self_consistent_field.jl

```diff
@@ -130,7 +133,7 @@ Solve the Kohn-Sham equations with a SCF algorithm, starting at ρ.
     # Callback is run one last time with final state to allow callback to clean up
     info = (ham=ham, basis=basis, energies=energies, converged=fpres.converged,
             ρ=ρout, eigenvalues=eigenvalues, occupation=occupation, εF=εF,
-            n_iter=n_iter, ψ=ψ, stage=:finalize)
+            n_iter=n_iter, ψ=ψ, diagonalization=info.diagonalization, stage=:finalize)
```

Comment:
Diagonalization_info? We probably need a more consistent naming for nested named tuples. Eg args for input and info for output, so this could be diag_info and the input parameters diag_args